### ISSUE TITLE
\tcboxmath does not work anymore inside Display Math

### ISSUE THREAD

**Issue by brunoh2015** [T-F-S/tcolorbox#162]
I have posted a detailed explanation at
[https://tex.stackexchange.com/questions/627387/tcboxmath-not-working-as-before-in-new-version-5-0-0-of-tcolorbox?noredirect=1#comment1564713_627387](url)

**Comment by u-fischer** [comment]
A failing MWE is

~~~~
\documentclass{article}
\usepackage[theorems, breakable]{tcolorbox}
\tcbset{breakable}

\begin{document}
Math Formula


\[\tcboxmath{x}\]

\end{document}

~~~~

It fails with

~~~~
! Missing $ inserted.
<inserted text> 
                $
l.118 \[\tcboxmath{x}
                     \]
? x
~~~~

**Comment by muzimuzhi** [comment]
Caused by the breakable part of https://github.com/T-F-S/tcolorbox/commit/11e2ba1bfe76ab1efb7bef77d46b0d9a109437f6 which tries to fix #157.

After that commit, in `\tcb@ox`, 
 - `\tcb@apply@box@options{#1}` first sets `\tcbset{#1}` then uses `\iftcb@isbreakable`;
 - then `\tcbox@inner@(hbox|minipage)` is used in which option `breakable@false` is set, and that's too late.

https://github.com/T-F-S/tcolorbox/blob/b0e2c27bcdbc3fa3eb830200aa55c81e9ce7bc5b/tex/latex/tcolorbox/tcolorbox.sty#L1904-L1915

**Comment by T-F-S** [comment]
I'm preparing a fix with version `5.0.1` (in production). @muzimuzhi is correct in the analysis of the problem - I move the setting of `breakable@false` to an earlier and more appropriate place.

**Comment by brunoh2015** [comment]
everything is back to normal now!
